querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java

```diff
@@ -895,13 +895,7 @@ protected void visitOperation(Class<?> type, Operator operator, List<? extends E
             }
         }
 
-        if (operator == Ops.LIST && args.get(0) instanceof SubQueryExpression) {
-            boolean oldUnion = inUnion;
-            inUnion = true;
-            super.visitOperation(type, SQLOps.UNION, args);
-            inUnion = oldUnion;
-
-        } else if (operator == SQLOps.UNION || operator == SQLOps.UNION_ALL) {
+        if (operator == SQLOps.UNION || operator == SQLOps.UNION_ALL) {
```

Comment:
The problem is that the LIST operator is too general. It was used to turn

ex in (subquery1, subquery2, ...)

into

ex in (subquery1 union subquery2 union ...)

One way to fix it would be to introduce a SET operator and use that in ex in args expressions.